activitysim/abm/models/telecommute_status.py

```diff
@@ -0,0 +1,125 @@
+# ActivitySim
+# See full license in LICENSE.txt.
+from __future__ import annotations
+
+import logging
+
+import numpy as np
+import pandas as pd
+
+from activitysim.core import (
+    config,
+    estimation,
+    expressions,
+    simulate,
+    tracing,
+    workflow,
+)
+from activitysim.core.configuration.base import PreprocessorSettings, PydanticReadable
+from activitysim.core.configuration.logit import LogitComponentSettings
+
+logger = logging.getLogger("activitysim")
+
+
+class TelecommuteStatusSettings(LogitComponentSettings, extra="forbid"):
+    """
+    Settings for the `telecommute_status` component.
+    """
+
+    preprocessor: PreprocessorSettings | None = None
+    """Setting for the preprocessor."""
+
+    TELECOMMUTE_ALT: int
+    """Value that specifies if the worker is telecommuting on the simulation day."""
+
+    CHOOSER_FILTER_COLUMN_NAME: str = "is_worker"
+    """Column name in the dataframe to represent worker."""
+
+
+@workflow.step
+def telecommute_status(
+    state: workflow.State,
+    persons_merged: pd.DataFrame,
+    persons: pd.DataFrame,
+    model_settings: TelecommuteStatusSettings | None = None,
+    model_settings_file_name: str = "telecommute_status.yaml",
+    trace_label: str = "telecommute_status",
+) -> None:
+    """
+    This model predicts whether a person (worker) telecommutes on the simulation day.
+    The output from this model is TRUE (if telecommutes) or FALSE (if does not telecommute).
+    """
+    if model_settings is None:
+        model_settings = TelecommuteStatusSettings.read_settings_file(
+            state.filesystem,
+            model_settings_file_name,
+        )
+
+    choosers = persons_merged
+    chooser_filter_column_name = model_settings.CHOOSER_FILTER_COLUMN_NAME
+    choosers = choosers[(choosers[chooser_filter_column_name])]
+    logger.info("Running %s with %d persons", trace_label, len(choosers))
+
+    estimator = estimation.manager.begin_estimation(state, "telecommute_status")
+
+    constants = config.get_model_constants(model_settings)
+
+    # - preprocessor
+    preprocessor_settings = model_settings.preprocessor
+    if preprocessor_settings:
+        locals_d = {}
+        if constants is not None:
+            locals_d.update(constants)
+
+        expressions.assign_columns(
+            state,
+            df=choosers,
+            model_settings=preprocessor_settings,
+            locals_dict=locals_d,
+            trace_label=trace_label,
+        )
```

Comment:
```suggestion
     expressions.annotate_preprocessors(
          state,
          df=choosers,
          locals_dict={} if constants is None else constants,
          skims=None,
          model_settings=model_settings,
          trace_label=trace_label,
      )
        
```
Update with the new preprocessor function call

Reply:
Cool, thank you, I will also add the post-model annotator call.